controllers/ratelimitpolicy_enforced_status_controller.go

```diff
@@ -177,7 +177,7 @@ func (r *RateLimitPolicyEnforcedStatusReconciler) hasErrCondOnSubResource(ctx co
 
 func (r *RateLimitPolicyEnforcedStatusReconciler) setCondition(ctx context.Context, logger logr.Logger, p *kuadrantv1beta2.RateLimitPolicy, conditions *[]metav1.Condition, cond metav1.Condition) error {
 	idx := utils.Index(*conditions, func(c metav1.Condition) bool {
-		return c.Type == cond.Type && c.Status == cond.Status && c.Reason == cond.Reason && c.Message == cond.Message
+		return c.Type == cond.Type && c.Status == cond.Status && c.Reason == cond.Reason
```

Comment:
Testing the status message here was causing infinite loop of status reconciliation when a route policy is overridden by more than one gateway policy.

Reply:
I've fixed this so the message is comparable too.